### ISSUE TITLE
[unbound-method] rule triggered when binding a class method

### ISSUE THREAD

**Issue by kai670** [typescript-eslint/typescript-eslint#700]
**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/unbound-method": ["error"]
  }
}
```

```TS
class Foo {
	constructor() {
		this.onMessage = this.onMessage.bind(this);
	}

	onMessage(): void {}
}
```

**Expected Result**
No error detected

**Actual Result**
`Avoid referencing unbound methods which may cause unintentional scoping of `this` @typescript-eslint/unbound-method`

**Additional Info**
As far as I know this is a common pattern for binding event handlers. Could be related with #692

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `1.11.0` |
| `@typescript-eslint/parser`        | `1.11.0` |
| `TypeScript`                       | `3.5.3` |
| `ESLint`                           | `5.16.0` |
| `node`                             | `8.16.0` |
| `npm`                              | `6.10.0` |

**Comment by glen-84** [comment]
Is this a duplicate of #636?
